src/Aspire.Dashboard/Model/DashboardClient.cs

```diff
@@ -98,6 +103,28 @@ GrpcChannel CreateChannel()
                 KeepAlivePingPolicy = HttpKeepAlivePingPolicy.WithActiveRequests
             };
 
+            if (!configuration.GetBool(ResourceServiceDisableAuthVariableName, defaultValue: false))
+            {
+                // Auth hasn't been suppressed, so configure it.
+                var certificatePath = configuration[ResourceServiceClientCertificatePathVariableName];
+                var certificatePassword = configuration[ResourceServiceClientCertificatePasswordVariableName];
+
+                if (string.IsNullOrWhiteSpace(certificatePath))
+                {
+                    throw new InvalidOperationException(
+                        $"Resource service auth hasn't been disabled via {ResourceServiceDisableAuthVariableName}, " +
+                        $"and no certificate was provided via {ResourceServiceClientCertificatePathVariableName} and " +
+                        $"{ResourceServiceClientCertificatePasswordVariableName}. Review the dashboard's configuration.");
+                }
+
+                httpHandler.SslOptions = new SslClientAuthenticationOptions
```

Comment:
We may wish to have a `RemoteCertificateValidationCallback` here that validates the server certificate's thumbprint (and corresponding config to pass that cert in).